src/Microsoft.Health.Fhir.Core/Features/KnownHeaders.cs

```diff
@@ -31,7 +31,7 @@ public static class KnownHeaders
         public const string CustomAuditHeaderPrefix = "X-MS-AZUREFHIR-AUDIT-";
         public const string FhirUserHeader = "x-ms-fhiruser";
 
-        // #conditionalQueryParallelism - Header used to activate parallel conditional-query processing.
-        public const string ConditionalQueryProcessingLogic = "x-conditionalquery-processing-logic";
+        // #maxQueryParallelism - Header used to activate parallel conditional-query processing.
+        public const string QueryProcessingLogic = "x-ms-query-processing-logic";
```

Comment:
Its a conventions thats hard to break :D but with the link above: https://datatracker.ietf.org/doc/html/rfc6648

I know the naming is supposed to be generic, but could it be something like "x-ms-optimize-query-logic"?

Really, the purpose is to try to reduce latency at the cost of RUs... `x-ms-query-latency-over-efficiency`?